### PR TITLE
tcl: Un-vendor sqlite, properly fix apple shared install names

### PR DESCRIPTION
**Pull request by Ahajha**
Specify library name and version:  **tcl/all**

Tcl vendors sqlite3 for an extension, and it seems this had gone unnoticed until now. This also adds an option to remove sqlite entirely in case users don't want this.

Tcl has a lot of sub-directories with libraries in the package folder, and the `fix_apple_shared_install_name` tool wasn't fixing them. Adding the correct folders to `self.cpp.package.libdirs` allows it to fix everything.

Blocks https://github.com/conan-io/conan-center-index/pull/21387

---

- [X] I've read the [contributing guidelines](https://github.com/conan-io/conan-center-index/blob/master/CONTRIBUTING.md).
- [X] I've used a [recent](https://github.com/conan-io/conan/releases/latest) Conan client version close to the [currently deployed](https://github.com/conan-io/conan-center-index/blob/master/.c3i/config_v1.yml#L6).
- [X] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Review by RubenRBS — COMMENTED** [review]
I think this works, but will delay approving until tomorrow, @uilianries and I want to use this PR for some internal training, sorry about the delay :)

File: recipes/tcl/all/conanfile.py [hunk 1]
```diff
@@ -98,6 +120,26 @@ def yes_no(v): return "yes" if v else "no"
...
+            if self.options.with_sqlite:
+                # tdbcsqlite is the wrapper package for tcl, don't modify it
+                if "sqlite" in folder and "tdbc" not in folder:
+                    rm(self, "*.c", os.path.join(self.source_folder, "pkgs", folder, "compat"), recursive=True)
+                    rm(self, "*.h", os.path.join(self.source_folder, "pkgs", folder, "compat"), recursive=True)
+
+                    # Don't override SQLITE_* variables, causes link errors
+                    if Version(self.version) < "8.6.13":
+                        replace_in_file(self, os.path.join(self.source_folder, "pkgs", folder, "Makefile.in"), "-DSQLITE_API=MODULE_SCOPE", "")
+                    replace_in_file(self, os.path.join(self.source_folder, "pkgs", folder, "Makefile.in"), "-DSQLITE_EXTERN=", "")
```
Comment (RubenRBS) [inline recipes/tcl/all/conanfile.py:135]: Can you give me some insight into these changes? I haven't been able to find the compilation logs for the failed case. Thanks :)

**Review by RubenRBS — COMMENTED** [review]
Thanks a lot! I have a question regarding the compilation issues.

Also, is there a reason we can't accept the vendorization? Most (if not all) package managers seem to also vendorize sqlite for this library. What is exactly the blocker here? (Did you get a chance to report this to upstream so they can unvendorize it by themselves?)
Thanks :)

**Comment by Ahajha** [comment]
> Thanks a lot! I have a question regarding the compilation issues.
> 
> Also, is there a reason we can't accept the vendorization? Most (if not all) package managers seem to also vendorize sqlite for this library. What is exactly the blocker here? (Did you get a chance to report this to upstream so they can unvendorize it by themselves?) Thanks :)

It wasn't specifically a problem, but I figured it shouldn't be vendored on principle. Wasn't aware that this was "allowed" to some extent.

I doubt they'd accept such a patch, they actually have a note that while they have an option to use an external sqlite, they don't recommend it. So I imagine this is very intentional.

File: recipes/tcl/all/conanfile.py (hunk 1 above)
Comment (Ahajha) [reply recipes/tcl/all/conanfile.py:135]: As far as I can tell, it messes with SQLite's linkage. These are overridden in TCL's build, so these are just un-overriding it to the defaults.

**Comment by Ahajha** [comment]
The main issue was that when CPython needed to link against tcl(shared), the rpaths were not correctly fixed on Mac. However, based on your questions, I'm now debating if maybe this could be fixed in a different way:

Perhaps we could keep the vendoring, but I don't think we need to list the extra libraries that are only used for the TCL extensions. So packages that link against it should only use the main TCL library. That would basically just simplify the package info. I'll make another PR in a bit to consider.

**Comment by Ahajha** [comment]
@RubenRBS Sorry for the delay, I've opened an alternate PR that I think fixes the problem in a better way: https://github.com/conan-io/conan-center-index/pull/22939. This can probably be closed, but I'll leave it open in case you think otherwise. The changes here will be pointless if the other PR is merged.

**Comment by Ahajha** [comment]
#22939 was merged, this is no longer necessary.

**Pull request closed without merging**